### PR TITLE
Make meta-coral compatible with ISAR-sbuild - v2

### PR DESCRIPTION
**Pull request by fmoessbauer**

### CODE REVIEW

**Comment by fmoessbauer** [comment]
@chombourger , @jan-kiszka This MR supersedes #22 (it basically is a combination of #22 and #23). Please review.
I locally tested that on a not too trivial layer and it worked.

**Comment by chombourger** [comment]
thanks! will definitely take a look today

File: classes/dpkg-bazel.bbclass [hunk 1]
```diff
@@ -12,20 +12,30 @@
...
+# ccache + bazel is broken
+USE_CCACHE="0"
 
 # fetch all bazel deps and add to package sources
 bazel_fetch() {
-    dpkg_do_mounts
     E="${@ isar_export_proxies(d)}"
-    sudo chroot --userspec=$( id -u ):$( id -g ) ${BUILDCHROOT_DIR} /bin/bash -c 'cd ${PP}/${PPS} && rm -rf local && ./debian/rules local'
-    dpkg_undo_mounts
+    echo "SCHROOT: ${SBUILD_CHROOT}"
```
Comment (chombourger) [inline classes/dpkg-bazel.bbclass]: looks like a debug print, remove?

File: classes/dpkg-bazel.bbclass [hunk 2]
```diff
@@ -12,20 +12,30 @@
...
+    schroot -d / -c ${SBUILD_CHROOT} -u root -- sh <<EOF
+        echo ${ISAR_APT_REPO} > /etc/apt/sources.list.d/isar_apt.list;
```
Comment (chombourger) [inline classes/dpkg-bazel.bbclass:33]: do we need to start the inline script with `set -e` to abort on the first error?

Comment (fmoessbauer) [reply classes/dpkg-bazel.bbclass:33]: Yes. good catch.

File: classes/dpkg-bazel.bbclass [hunk 3]
```diff
@@ -12,20 +12,30 @@
...
+                    --no-install-recommends --allow-downgrades \
+                    install ${BAZEL_FETCH_DEPENDS}
+        cd ${PP}/${PPS} && ./debian/rules local
+EOF
 
     # generate orig archive, compress in parallel
-    tar -c -I 'xz -1 -T0' -f ${WORKDIR}/${PN}_${PV}.orig.tar.xz -C ${S} --exclude=.git --exclude='./debian' .
-    # fix permissions of bazel cache
-    sudo chown -R $( id -u ):$( id -g ) ${WORKDIR}/.cache/bazel
+    tar -c -I 'xz -1 -T0' -f ${WORKDIR}/${PN}_${PV}.orig.tar.xz -C ${S} --exclude=.git --exclude=.env .
```
Comment (fmoessbauer) [inline classes/dpkg-bazel.bbclass]: We should add xz options similar to the ones from `imagetypes.bbclass` to avoid OOM in CI.

**Comment by chombourger** [comment]
Other than the few things I noted, this PR is IMHO good to go. thanks @fmoessbauer !

File: classes/dpkg-bazel.bbclass (hunk 1 above)
Comment (fmoessbauer) [reply classes/dpkg-bazel.bbclass]: Removed.

**Review by chombourger — APPROVED** [review]

**Pull request merged**